### PR TITLE
Fix polkit configuration issue in split daemon job

### PR DESCRIPTION
**Pull request by Yingshun**
In split daemon environment, it needs to modify polkit related
settings in virtqemud.conf.

depends on https://github.com/avocado-framework/avocado-vt/pull/2646
Signed-off-by: Yingshun Cui <yicui@redhat.com>

### CODE REVIEW

**Comment by Yingshun** [comment]
**Test result:**
_split daemon env:_
```
(.libvirt-ci-venv-ci-runtest-SYSx7x) [root@dell-per6415-01 network]# avocado run --vt-type libvirt virsh.net_define_undefine.normal_test.new_network.acl_test 
JOB ID     : e65bfaa1d6348046710775129c8e4ce4d44fc9e5
JOB LOG    : /root/avocado/job-results/job-2020-07-06T02.25-e65bfaa/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.normal_test.new_network.acl_test: PASS (8.99 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 10.75 s

```
_non split daemon env:_
libvirt 4.5
```
#  avocado run --vt-type libvirt virsh.net_define_undefine.normal_test.new_network.acl_test 
JOB ID     : 21e7410d9f986cc20417f43b98f3ef6a286b21f9
JOB LOG    : /root/avocado/job-results/job-2020-07-06T02.22-21e7410/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.normal_test.new_network.acl_test: PASS (10.25 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 12.52 s
```

libvirt 6.0:
```
# avocado run --vt-type libvirt virsh.net_define_undefine.normal_test.new_network.acl_test 

JOB ID     : 7f0a4318409cdb1975afdd448f1bd44cd8cd136b
JOB LOG    : /root/avocado/job-results/job-2020-07-06T02.36-7f0a431/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.net_define_undefine.normal_test.new_network.acl_test: PASS (12.81 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 14.26 s

```

File: virttest/env_process.py [hunk 1]
```diff
@@ -1297,6 +1303,9 @@ def preprocess(test, params, env):
             if libvirtd_inst is None:
                 libvirtd_inst = utils_libvirtd.Libvirtd()
             libvirtd_inst.restart()
+            if virtqemud_inst is None:
```
Comment (chunfuwen) [inline virttest/env_process.py:1306]: you may add one factory method in utils_split_daemons.py
    def create_daemon_instance():
        if libvirtd_decorator.get_libvirt_version_compare(5, 6, 0) and
               libvirtd_decorator.get_libvirtd_split_enable_bit():
            return utils_split_daemons.VirtQemud()
        else:
            return  utils_libvirtd.Libvirtd()

libvirtd_inst = create_daemon_instance()
libvirtd_inst.restart()

File: virttest/env_process.py [hunk 2]
```diff
@@ -1295,7 +1300,7 @@ def preprocess(test, params, env):
             except Exception as e:
                 logging.error("Unexpected error: '%s'" % str(e))
             if libvirtd_inst is None:
-                libvirtd_inst = utils_libvirtd.Libvirtd()
+                libvirtd_inst = utils_libvirtd.Libvirtd("virtqemud")
```
Comment (kylazhang) [inline virttest/env_process.py:1303]: I think here better remove the "virtqemud" to be more clear

Comment (Yingshun) [reply virttest/env_process.py:1303]: For modular daemon env, the configuration file is changed from libvirtd.conf to virtqemud.conf, so I add the specific daemon name here.
utils_libvirtd.Libvirtd() with virtqemud will create an object which can restart virtqemud or libvirtd. If there is no "virtqemud", it will restart all Virt*d or libvirtd. Anyway, each of them can work well. let's wait for others' feedback if they have same problem.  @yafu-1 As feature owner, do you have any suggestions?

File: virttest/env_process.py [hunk 3]
```diff
@@ -1285,6 +1287,9 @@ def preprocess(test, params, env):
         # work on connect_uri as default behavior.
         os.environ['LIBVIRT_DEFAULT_URI'] = connect_uri
         if params.get("setup_libvirt_polkit") == "yes":
+            if (libvirtd_ver.version_compare(5, 6, 0) and
+               libvirtd_decorator.get_libvirtd_split_enable_bit()):
```
Comment (dzhengfy) [inline virttest/env_process.py:1291]: I suggest we move L1290 L1291 together in near feature. But we can do it in another PR as there are many  places like this.

**Comment by dzhengfy** [comment]
Waiting for #2646 to be merged.

File: virttest/env_process.py (hunk 2 above)
Comment (chunfuwen) [reply virttest/env_process.py:1303]: As long evolution solution, utils_libvirtd.Libvirtd module need step away, I don't suggest use  utils_libvirtd.Libvirtd("xx") to create new split daemon, such as virtqemud and virtproxyd.

**Comment by chunfuwen** [comment]
@santwana 
@smitterl 
@clebergnu 

If you can provide your ideas from your expertise, we will be very grateful

File: virttest/env_process.py (hunk 2 above)
Comment (Yingshun) [reply virttest/env_process.py:1303]: @chunfuwen I can see why you don't like it. But I hope you understand that this solution is not my personal decision, but the result of the team discussion. This is the way we hope to see at this stage. You've added label to request 2 maintainers and also requested outside maintainers, unfortunately, there was few response. Anyway, what I want to say is we've been waiting for enough time, it's time to move on.

Comment (chunfuwen) [reply virttest/env_process.py:1303]: @Yingshun, as I mentioned before, this is our final evolution version. Therefore involve in more participants to review will help work out optimized solution. 
Moreover, as per feedback from feature team, the real requirement is not clear, so not sure whether it is right time to move it  forward now.

**Comment by dzhengfy** [comment]
As team's discussion, we decide to use current solution to conduct the initial test for Libvirt modular daemon. The current solution is to operate all sub-daemons at same time by default. But we also support operate a single sub daemon if a service name is passed. This solution can meet our current test requirement with the minimal coding effort.

**Comment by dzhengfy** [comment]
We will create new PRs for other enhancements in future.
